metric.go

```diff
@@ -56,7 +52,7 @@ type counterMetric struct {
 	p           *metric.Int64Counter
 }
 
-func newMetrics(cfg Config) (metrics, error) {
+func newMetrics(cfg Config) (*metrics, error) {
```

Comment:
Is this necessary? I think we should either revert this, or update the `Appender.metrics` field type to `*metrics`

Reply:
Oh good catch, it isn't necessary. It's leftover from the previous experiment. thanks!